dask/base.py

```diff
@@ -1397,14 +1396,16 @@ def get_scheduler(get=None, scheduler=None, collections=None, cls=None):
     if config.get("get", None):
         raise ValueError(get_err_msg)
 
-    if getattr(thread_state, "key", False):
-        from distributed.worker import get_worker
-
-        return get_worker().client.get
-
     if cls is not None:
         return cls.__dask_scheduler__
 
+    try:
+        from distributed import get_client
+
+        return get_client().get
+    except ValueError:
+        pass
```

Comment:
i.e. if we're not requesting *anything* specific but are executing the default we will always try to get a global distributed client before checking the collections default.
The precedence there is

1. Worker client
2. Current client

there is effectively no performance penalty in case there is no distributed worker or client other than raise+except